nixos/modules/programs/openvpn3.nix

```diff
@@ -109,7 +113,12 @@ in
     environment = {
       systemPackages = [ cfg.package ];
       etc = {
-        "openvpn3/netcfg.json".source = json.generate "netcfg.json" cfg.netcfg.settings;
+        "openvpn3/netcfg.json".source = json.generate "netcfg.json" (
+          let
+            netcfg = cfg.netcfg.settings;
+          in
+          if netcfg.systemd_resolved then netcfg else builtins.removeAttrs netcfg [ "systemd_resolved" ]
```

Comment:
```suggestion
          builtins.removeAttrs netcfg (optional !netcfg.systemd_resolved "systemd_resolved")
```

Reply:
By the way, is there a way to describe this via option definition?

Reply:
Not if you want systemd_resolved defined as an option, and not just freeform value.

Reply:
Thanks, updated